web/src/main/java/com/google/accompanist/web/WebView.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.web
+
+import android.graphics.Bitmap
+import android.net.Uri
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import androidx.activity.compose.BackHandler
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.viewinterop.AndroidView
+
+/**
+ * A wrapper around the Android View WebView to provide a basic WebView composable.
+ *
+ * If you require more customisation you are most likely better rolling your own and using this
+ * wrapper as an example.
+ *
+ * @param state The webview state holder where the Uri to load is defined.
+ * @param captureBackPresses Set to true to have this Composable capture back presses and navigate
+ * the WebView back.
+ * @param onCreated Called when the WebView is first created, this can be used to set additional
+ * settings on the WebView.
+ *
+ * @sample com.google.accompanist.sample.webview.BasicWebViewSample
+ */
+@Composable
+fun WebView(
+    state: WebViewState,
+    onContentChanged: (WebContent) -> Unit,
```

Comment:
Do we need this callback? Couldn't we set `state.content` directly and consumers can watch this if they need to react? This would also alleviate callers from havig to do `{ state.content = it }`

Reply:
+1

Reply:
Because I also wanted the url to be able to be changed from compose state outside the webview and it just felt like having the callback is how other APIs handle this

Reply:
Ignore that, I have updated it and it is much better. Will remove onContentChanged